.github/workflows/ci.yml

```diff
@@ -229,3 +229,66 @@ jobs:
 
       - name: Run tests
         run: pytest tests -n auto -v
+
+  test_against_latest_sentry:
+    name: Sentry-Relay Integration Tests
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Relay
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          profile: minimal
+          override: true
+
+      - uses: swatinem/rust-cache@v1
+        with:
+          key: ${{ github.job }}
+
+      - name: Cargo build
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --all-features
+
+      # Checkout Sentry and run integreation tests against latest Relay
+      - name: Checkout sentry
+        uses: actions/checkout@v2
+        with:
+          repository: "getsentry/sentry"
+          path: sentry
+
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.6
+
+      - name: Get pip cache dir
+        id: pip
+        run: |
+          pip install --upgrade pip wheel
+          echo "::set-output name=pip-cache-dir::$(pip cache dir)"
+
+      - name: Cache (pip)
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.pip.outputs.pip-cache-dir }}
+          key: sentry-deps-${{ hashFiles('**/requirements**.txt') }}
+          restore-keys: sentry-deps-
+
+      # A slightly modified version from upstream since I can't make it change directories
+      - name: Setup Sentry
```

Comment:
I would have not needed to fork the action unless I was able to change directories.
I believe `setup-sentry/actions.yml` could be converted into a script and remove the need for an action.
In general, it would also help decouple from GH actions.

I think we could avoid the Docker test image work if we use this approach instead:
<img width="1059" alt="image" src="https://user-images.githubusercontent.com/44410/124773357-e3d53580-df0a-11eb-8a01-6ed611f28d60.png">

Reply:
I think this is all CI-specific setup, so I think it's fine that it is coupled to GHA. I think *if* these actions were allowed to call anything else besides shell commands (e.g. `uses:` for other actions), actions would be better. *if* being key here :(

I was hoping that having Docker test images would cut down on the setup time, but I think you're right and we should proceed without it.